lib/topological_inventory/persister/workflow.rb

```diff
@@ -0,0 +1,183 @@
+require "topological_inventory/persister/logging"
+
+module TopologicalInventory
+  module Persister
+    class Workflow
+      include Logging
+
+      def initialize(persister, messaging_client, payload)
+        @persister        = persister
+        @messaging_client = messaging_client
+        # TODO(lsmola) we should be able to reconstruct the payload out of persistor? E.g. to just repeat the sweep phase
+        @payload = payload
+      end
+
+      def execute!
+        if total_parts
+          sweep_inactive_records!
+        else
+          persist_collections!
+        end
+      end
+
+      private
+
+      attr_reader :persister, :messaging_client, :payload
+      delegate :inventory_collections,
+               :manager,
+               :persist!,
+               :refresh_state_uuid,
+               :refresh_state_part_uuid,
+               :total_parts,
+               :sweep_scope,
+               :to => :persister
+
+      def define_refresh_state_ics
+        refresh_states_inventory_collection = ::InventoryRefresh::InventoryCollection.new(
```

Comment:
These should just be defined in https://github.com/ManageIQ/topological_inventory-core/tree/master/lib/topological_inventory/schema/default.rb

Reply:
ah, right, these should not be part of refresh. I use these because I don't have upsert query abstracted enough. And I need to do upsert, so this works.

Reply:
Oh okay so they won't be inventory collections long term?

Reply:
yes, these should be just 2 upserts